### ISSUE TITLE
test_quic_multistream fails in nightly CI runs

### ISSUE THREAD

**Issue by ifranzki** [openssl/openssl#21242]
In our CI the OpenSSL tests run in a nightly fashion on the s390x platform.
Since 3 days the test_quic_multistream testcase fails sporadically as follows:
```
        # INFO:  @ test/quic_multistream_test.c:2008
        # Running script 18 (order=0)
        # ERROR: (uint64_t) '(uint64_t)rxke >= op->arg2' failed @ test/quic_multistream_test.c:314
        # [1] compared to [4]
        # ERROR: (bool) 'ok == true' failed @ test/quic_multistream_test.c:824
        # false
        # ERROR:  @ test/quic_multistream_test.c:1344
        # failed at script op 12, thread -1
        # 
        # 
        # ERROR: (bool) 'run_script_worker(&h, script, -1) == true' failed @ test/quic_multistream_test.c:1367
        # false
        # OPENSSL_TEST_RAND_ORDER=1687117718
        not ok 35 - iteration 35
# ------------------------------------------------------------------------------70-test_sslsignature.t .................. ok

    # OPENSSL_TEST_RAND_ORDER=1687117718
    not ok 1 - test_script
# ------------------------------------------------------------------------------
../../util/wrap.pl ../../test/quic_multistream_test ../../test/certs/servercert.pem ../../test/certs/serverkey.pem => 1
not ok 1
# ------------------------------------------------------------------------------
#   Failed test at test/recipes/70-test_quic_multistream.t line 19.
# Looks like you failed 1 test of 1.70-test_quic_multistream.t .............. 
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/1 subtests 
```
I don't see anything platform specific here. However, we run with `make HARNESS_JOBS=$(nproc) test` so maybe parallelism is the cause of the problem here.

**Comment by t8m** [comment]
There are also (even more sporadic) failures in the script 13.

**Comment by t8m** [comment]
@hlandau any ideas?

**Comment by t8m** [comment]
https://github.com/openssl/openssl/actions/runs/5356399498/jobs/9715879661
https://github.com/openssl/openssl/actions/runs/5272789536/jobs/9535469563

**Comment by hlandau** [comment]
Partial fix in #21310

**Comment by t8m** [comment]
Let's close this, we have https://github.com/openssl/project/issues/212 for the currently remaining problem with this test.
